### ISSUE TITLE
Deno panic in language server

### ISSUE THREAD

**Issue by mikecasinghino** [denoland/deno#25395]
```Platform: macos x86_64
Version: 1.46.2
Args: ["/Users/mc/.deno/bin/deno", "lsp"]

thread '<unnamed>' panicked at cli/lsp/diagnostics.rs:741:46:
called `Result::unwrap()` on an `Err` value: ParseError { index: 41, kind: UnexpectedChar }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 9:23:03 AM] The Deno Language Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

**Comment by mikecasinghino** [comment]
Looks like this is the real problem:

```
Could not convert URL "file:///Users/mc/wrk/wiki/wiki4/routes/p/[...path].tsx" to URI: unexpected character at index 41
Unable to fix lint error: unexpected character at index 41
successfully cancelled request with ID: 11
```

**Comment by nathanwhit** [comment]
This may have been fixed by https://github.com/denoland/deno/pull/25238

**Comment by nayeemrmn** [comment]
It's turned into a printed error now but still a bug. Fixed by #25509.
